### ISSUE TITLE
Pushes - Files and or Passwords are not removed

### ISSUE THREAD

**Issue by kayvanaarssen** [pglombardo/PasswordPusher#1689]
Seems like Files / Password Pushes when logged in are not removed after the Set Days / Views.
Is there a reason behind this? For example to view the Audit Logs or something?
They cannot be viewed anymore, but are still visible in the backend.

Would maybe be better to move this to a view with "Archived / Expired" Items
And leave the main page for the "Active" Shared items?

![image](https://github.com/pglombardo/PasswordPusher/assets/672841/32522111-bd66-4a29-9535-6b488618fb70)

**Comment by pglombardo** [comment]
Hi @kayvanaarssen - That is true.  To address, you can periodically run these [background tasks](https://github.com/pglombardo/PasswordPusher/wiki/Periodic-Background-Tasks) to pre-emptively expire pushes.

From that link:

> When a user requests to see a secret URL of an unexpired push, the application first validates if that password should be expired by re-validating the expiration settings. If the push is past it's expiration, the push payload is deleted from the database and the user gets the "This secret link has expired" message. The daily_expiration task serves to expire passwords preemptively which saves on some CPU and database calls ahead of time.

So the default is lazy expiration (expire on next view attempt) unless you run those jobs.

There are more reasons as well.  For example text based pushes [get expired automatically](https://github.com/pglombardo/PasswordPusher/blob/master/app/controllers/passwords_controller.rb#L96-L97) on last view but file pushes cannot for the reason [described in the comment here](https://github.com/pglombardo/PasswordPusher/blob/master/app/controllers/file_pushes_controller.rb#L94-L98).

So tldr; if you prefer, setup a cron job to periodically run those background jobs in your container.

Does this help?

**Comment by kayvanaarssen** [comment]
Sure, thought this was done by Default.
We run in Docker, so i'll setup a job to do this thanks!

**Comment by kayvanaarssen** [comment]
@pglombardo Cannot figure out how to run it, can you give me a pointer?
Just running: 

```bash
rails daily_expiration
rails delete_expired_and_anonymous
```
in docker container does not work :-D What am I missing here?

**Comment by pglombardo** [comment]
@kayvanaarssen I'll close out this issue.  If anything remains, please let me know.
